Suggestion: mark fixable checks in the default output?

One thing IMO that would really be nice would be Ruby Rubocop's feature of marking changes that support autofix. For example, on the dummy Ruby file:

```rb
# frozen_string_literal: true

def abs(x)
  if x > 0
    x
  else
    -x
  end
end
```

Rubocop produces this:

```
Inspecting 1 file
C

Offenses:

tmp.rb:3:9: C: Naming/MethodParameterName: Method parameter must be at least 3 characters long.
def abs(x)
        ^
tmp.rb:4:6: C: [Correctable] Style/NumericPredicate: Use x.positive? instead of x > 0.
  if x > 0
     ^^^^^

1 file inspected, 2 offenses detected, 1 offense auto-correctable
```

Yeah, the default is a little verbose, but notice the `[Correctable]` marker. That's fantastic, as it gives you an idea of what adding the fix (`-A` in Rubocop) flag will do, and what it won't. I miss not having some little mark or something in Ruff's output that lets me know what fixing will do and what it won't.

This is different than looking up each issue and checking to see if it's fixable or not, and it's a bit more helpful than just the summary of how many would be fixable, which is already there. I don't think it needs to be spelled out, but a little marker might be useful (like `[*]`, then `[*]` could be placed by the total number of fixable values, to help a user realize what it is there for.

Maybe the target is to try to mimic flake8's output as closely as possible, and that's okay, but just thought I'd point it out.

A few options...

```console
resources/test/fixtures/pyupgrade/UP035.py:47:14: F811 Redefinition of unused `Counter` from line 33
resources/test/fixtures/pyupgrade/UP035.py:50:15: F401 `a.b` imported but unused
resources/test/fixtures/pyupgrade/future_annotations.py:8:5: F401 `models.Nut` imported but unused
resources/test/fixtures/pyupgrade/future_annotations.py:26:19: F821 Undefined name `Bar` [*]
...
[*] 1 potentially fixable...
```

```console
resources/test/fixtures/pyupgrade/UP035.py:47:14: F811 Redefinition of unused `Counter` from line 33
resources/test/fixtures/pyupgrade/UP035.py:50:15: F401 `a.b` imported but unused
resources/test/fixtures/pyupgrade/future_annotations.py:8:5: F401 `models.Nut` imported but unused
[*] resources/test/fixtures/pyupgrade/future_annotations.py:26:19: F821 Undefined name `Bar`
...
[*] 1 potentially fixable...
```

I suspect that the former is more likely to be backwards-compatible for anyone relying on the format, and it avoids an offset in the filename.

Or:

```
resources/test/fixtures/pyupgrade/UP035.py:47:14: F811 Redefinition of unused `Counter` from line 33
resources/test/fixtures/pyupgrade/UP035.py:50:15: F401 `a.b` imported but unused
resources/test/fixtures/pyupgrade/future_annotations.py:8:5: F401 `models.Nut` imported but unused
resources/test/fixtures/pyupgrade/future_annotations.py:26:19: F821 [*] Undefined name `Bar`
```

Or

```
resources/test/fixtures/pyupgrade/UP035.py:47:14: F811 Redefinition of unused `Counter` from line 33
resources/test/fixtures/pyupgrade/UP035.py:50:15: F401 `a.b` imported but unused
resources/test/fixtures/pyupgrade/future_annotations.py:8:5: F401 `models.Nut` imported but unused
resources/test/fixtures/pyupgrade/future_annotations.py:26:19: [*] F821 Undefined name `Bar`
```

I like it at the end or in the middle (either example above), I don't like it at the beginning, makes the paths harder to read & quickly jump to.

I'm gonna go with:

```
resources/test/fixtures/pyupgrade/UP035.py:47:14: F811 Redefinition of unused `Counter` from line 33
resources/test/fixtures/pyupgrade/UP035.py:50:15: F401 `a.b` imported but unused
resources/test/fixtures/pyupgrade/future_annotations.py:8:5: F401 `models.Nut` imported but unused
resources/test/fixtures/pyupgrade/future_annotations.py:26:19: F821 [*] Undefined name `Bar`
```

It's least likely to break any existing clients as it retains the {file}:{row}:{column}:{code}:{message} structure (if we call `[*]` part of the message), visually clear, etc.

--- a/ruff_cli/src/printer.rs
+++ b/ruff_cli/src/printer.rs
@@ -122,7 +122,7 @@ impl<'a> Printer<'a> {
                         if num_fixable > 0 {
                             writeln!(
                                 stdout,
-                                "{num_fixable} potentially fixable with the --fix option."
+                                "[*] {num_fixable} potentially fixable with the --fix option."
                             )?;
                         }
                     }
@@ -475,17 +475,31 @@ fn num_digits(n: usize) -> usize {
 
 /// Print a single `Message` with full details.
 fn print_message<T: Write>(stdout: &mut T, message: &Message) -> Result<()> {
-    let label = format!(
-        "{}{}{}{}{}{} {} {}",
-        relativize_path(Path::new(&message.filename)).bold(),
-        ":".cyan(),
-        message.location.row(),
-        ":".cyan(),
-        message.location.column(),
-        ":".cyan(),
-        message.kind.rule().code().red().bold(),
-        message.kind.body(),
-    );
+    let label = if message.kind.fixable() {
+        format!(
+            "{}{}{}{}{}{} {} [*] {}",
+            relativize_path(Path::new(&message.filename)).bold(),
+            ":".cyan(),
+            message.location.row(),
+            ":".cyan(),
+            message.location.column(),
+            ":".cyan(),
+            message.kind.rule().code().red().bold(),
+            message.kind.body(),
+        )
+    } else {
+        format!(
+            "{}{}{}{}{}{} {} {}",
+            relativize_path(Path::new(&message.filename)).bold(),
+            ":".cyan(),
+            message.location.row(),
+            ":".cyan(),
+            message.location.column(),
+            ":".cyan(),
+            message.kind.rule().code().red().bold(),
+            message.kind.body(),
+        )
+    };
     writeln!(stdout, "{label}")?;
     if let Some(source) = &message.source {
         let commit = message.kind.commit();
@@ -540,16 +554,29 @@ fn print_grouped_message<T: Write>(
     row_length: usize,
     column_length: usize,
 ) -> Result<()> {
-    let label = format!(
-        "  {}{}{}{}{}  {}  {}",
-        " ".repeat(row_length - num_digits(message.location.row())),
-        message.location.row(),
-        ":".cyan(),
-        message.location.column(),
-        " ".repeat(column_length - num_digits(message.location.column())),
-        message.kind.rule().code().red().bold(),
-        message.kind.body(),
-    );
+    let label = if message.kind.fixable() {
+        format!(
+            "  {}{}{}{}{}  {}  [*] {}",
+            " ".repeat(row_length - num_digits(message.location.row())),
+            message.location.row(),
+            ":".cyan(),
+            message.location.column(),
+            " ".repeat(column_length - num_digits(message.location.column())),
+            message.kind.rule().code().red().bold(),
+            message.kind.body(),
+        )
+    } else {
+        format!(
+            "  {}{}{}{}{}  {}  {}",
+            " ".repeat(row_length - num_digits(message.location.row())),
+            message.location.row(),
+            ":".cyan(),
+            message.location.column(),
+            " ".repeat(column_length - num_digits(message.location.column())),
+            message.kind.rule().code().red().bold(),
+            message.kind.body(),
+        )
+    };
     writeln!(stdout, "{label}")?;
     if let Some(source) = &message.source {
         let commit = message.kind.commit();

--- a/ruff_cli/tests/integration_test.rs
+++ b/ruff_cli/tests/integration_test.rs
@@ -31,8 +31,10 @@ fn test_stdin_error() -> Result<()> {
         .failure();
     assert_eq!(
         str::from_utf8(&output.get_output().stdout)?,
-        "-:1:8: F401 `os` imported but unused\nFound 1 error.\n1 potentially fixable with the \
-         --fix option.\n"
+        r#"-:1:8: F401 [*] `os` imported but unused
+Found 1 error.
+[*] 1 potentially fixable with the --fix option.
+"#
     );
     Ok(())
 }
@@ -54,8 +56,10 @@ fn test_stdin_filename() -> Result<()> {
         .failure();
     assert_eq!(
         str::from_utf8(&output.get_output().stdout)?,
-        "F401.py:1:8: F401 `os` imported but unused\nFound 1 error.\n1 potentially fixable with \
-         the --fix option.\n"
+        r#"F401.py:1:8: F401 [*] `os` imported but unused
+Found 1 error.
+[*] 1 potentially fixable with the --fix option.
+"#
     );
     Ok(())
 }

